internal/hcp/registry/json.go

```diff
@@ -83,7 +83,14 @@ func (h *JSONRegistry) PopulateVersion(ctx context.Context) error {
 
 // StartBuild is invoked when one build for the configuration is starting to be processed
 func (h *JSONRegistry) StartBuild(ctx context.Context, build sdkpacker.Build) error {
-	return h.bucket.startBuild(ctx, build.Name())
+	name := build.Name()
+
+	metadata := build.(*packer.CoreBuild).GetMetadata()
+	err := h.bucket.AddMetadataToBuild(ctx, name, metadata)
```

Comment:
Is `build` the same object that you get doing `buildToUpdate, err := bucket.Version.Build(buildName)`?